Board.java

```diff
@@ -13,15 +15,27 @@ public String getState() {
         return this.state;
     }
 
+     public Boolean availablePosition(int pos) {
+        if (state.charAt(pos) == '-' && pos >= 0 && pos <= dimension) {
```

Comment:
also if dimension is 9 then the condition should be:
position >=0 and position < dimension e.g. 0<=position <9